### DISCUSSION TITLE
QOL triggers

### DISCUSSION THREAD

**Discussion by Amidweiz** [ikemen-engine/Ikemen-GO#1598] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Here's a series of quality of life triggers I feel that the ikemen GO engine could use

### Describe the solution you'd like

IsProjectile
Returns non zero if a helper happens to be using `helpertype = projectile`

HasClsn()
Takes in a mandatory number based off of the collision number inside the air file and returns non zero if conditions have met

IsTrans
Returns non zero if the player has transparency applied to them

Angle
Returns the rotation angle value applied by AngleAdd, AngleDraw, AngleSet, and AngleMul state controllers

IsGravity
Returns non zero if the player is effected by the gravity state controller

### Additional context

`Helpertype = projectile` happens to be a valid undocumented valid helpertype in mugen however it doesn't do anything.
HasClsn would be handy because currently to detect if a character happens to have no blue hitboxes is utilizing a `helpertype = player` helper, spawning them at p2's position, and seeing if their position is the same as p2s which shouldn't be the case because of the player helper.

### Search terms

trigger, QOL

**Comment by potsmugen** [comment, marked as answer]
I think you're in luck. As of the current Nightly build:

> HasClsn()
> Takes in a mandatory number based off of the collision number inside the air file and returns non zero if conditions have met

AnimFrame(NumClsn1) and AnimFrame(NumClsn2) triggers.

> IsTrans
> Returns non zero if the player has transparency applied to them

Alpha Source and Alpha Dest triggers for transparency applied with Trans.
AnimFrame(AlphaSource) and AnimFrame(AlphaDest) for transparency defined in the AIR file.

> Angle
> Returns the rotation angle value applied by AngleAdd, AngleDraw, AngleSet, and AngleMul state controllers

Already exists exactly as you described it.

> IsGravity
> Returns non zero if the player is effected by the gravity state controller

I think this one's oddly specific. There's a Physics trigger however.

> IsProjectile
> Returns non zero if a helper happens to be using helpertype = projectile

I think a HelperType trigger would cover that and more.
